AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/OutgoingMutationQueue.swift

```diff
@@ -79,6 +79,10 @@ final class OutgoingMutationQueue: OutgoingMutationQueueBehavior {
     func startSyncingToCloud(api: APICategoryGraphQLBehavior,
                              mutationEventPublisher: MutationEventPublisher) {
         log.verbose(#function)
+        let outboxStatusEvent = OutboxStatusEvent(isEmpty: operationQueue.operationCount == 0 ? true : false)
+        let outboxStatusEventPayload = HubPayload(eventName: HubPayload.EventName.DataStore.outboxStatus,
+                                             data: outboxStatusEvent)
+        Amplify.Hub.dispatch(to: .dataStore, payload: outboxStatusEventPayload)
         stateMachine.notify(action: .receivedStart(api, mutationEventPublisher))
```

Comment:
So in iOS implementation. when DataStore starts and MutationEvent database is not empty, `requestEvent()` wil be called and then `enqueue()`, so that in `enqueue()` method, `dispatchOutboxStatusEvent(isEmpty: Bool)` already includes two scenarios: 1. when DataStore starts and 2. outboxMutationEnqueued. Is that the correct understanding?

Reply:
if enqueue() is called when datastore starts, yes the outbox mutation is enqueued, but this is when there are mutation events to be processed at datastore start. in OutgoingMutationQueue, there isn't a way to know when DataStore starts, if there are no mutation events. you only know when it is Not empty because your subscriber gets an mutation event from mutationEventPublisher